resources/check-layout-th.js

```diff
@@ -208,12 +208,21 @@ var testNumber = 0;
 var highlightError = false; // displays outline around failed test element.
 var printDomOnError = true; // prints dom when test fails.
 
-window.checkLayout = function(selectorList, callDone = true)
+window.checkLayout = async function(selectorList, callDone = true)
 {
     if (!selectorList) {
         console.error("You must provide a CSS selector of nodes to check.");
         return;
     }
+
+    // Allow for a delay before checking the layout
+    // Polyfills can set this variable to give themselves time to apply changes
+    const delay = window.CHECK_LAYOUT_DELAY_MS;
+    if (typeof delay === 'number') {
+        console.log(`Waiting ${delay}ms before checking layout...`);
+        await new Promise(r => setTimeout(r, delay));
```

Comment:
@mfreed7 [said](https://github.com/oddbird/wpt/commit/6f92b20ff9a16c159f93af655f468d29071becd6#r97831058):

> You'll want to read https://web-platform-tests.org/writing-tests/testharness-api.html#timers-in-tests and modify this code accordingly.

Reply:
@mfreed7 thanks, looking into that now

Reply:
Ok, I'm probably wrong, but isn't the problem running `setTimeout` inside a `test()` callback? In this case, the delay is applied before the first `test()` call is made inside the forloop, so the test body has no timers in it. I played around with using `t.step_wait()` and `t.step_timeout` inside the forloop, but I'm clearly doing something wrong because tests either complete with 0 assertions, or time out entirely 😕

@jgerigmeyer could you take a look at the link above and see if can think of a way to replace `setTimeout` with the helper methods? It would help to bounce ideas at some point tomorrow

Reply:
@jerivas I think you're right -- in this case we want a single delay before running the tests, not a delay inside each individual tests. It works for me to use the global version from the docs, e.g.:

```
await new Promise(r => step_timeout(r, delay));
```

I'm not sure that's functionally different than a plain `setTimeout` in this case, but 🤷

Reply:
Re @mfreed7's other comment...

> One comment on the code: don't you need to add **`await`** `checkLayout()` everywhere? Several of the tests call `checkLayout` twice, doing things between the calls.

I think that's correct -- we'd want to add `await checkLayout()`, at least any place where it's called twice in a given test.

Reply:
@jgerigmeyer thanks, `step_timeout` seems to do the trick, and it actually removes the need for `async` altogether. I ended up implementing it as a wrapper around the original `checkLayout` code